engine/access/rpc/backend/transactions/provider/execution_node.go

```diff
@@ -291,13 +354,54 @@ func (e *ENTransactionProvider) TransactionResultsByBlockID(
 			Events:        events,
 			ErrorMessage:  systemTxResult.GetErrorMessage(),
 			BlockID:       blockID,
-			TransactionID: e.systemTxID,
+			TransactionID: systemTxIDs[i],
 			BlockHeight:   block.Height,
 		})
 	}
+
 	return results, nil
 }
 
+// systemTransactionIDs determines the system transaction IDs upfront
+func (e *ENTransactionProvider) systemTransactionIDs(
+	systemTxResults []*execproto.GetTransactionResultResponse,
+	actualEventEncodingVersion entities.EventEncodingVersion,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+) ([]flow.Identifier, error) {
+	var systemTxIDs []flow.Identifier
+	scheduledCallbacksEnabled := len(systemTxResults) > 1 // TODO: improve this assumption
+
+	if !scheduledCallbacksEnabled {
+		return []flow.Identifier{e.systemTxID}, nil
+	}
+
+	// Get events from the process scheduled callback transaction to reconstruct the system collection
+	processResult := systemTxResults[0]
+	events, err := convert.MessagesToEventsWithEncodingConversion(
+		processResult.GetEvents(),
+		actualEventEncodingVersion,
+		requiredEventEncodingVersion,
+	)
+	if err != nil {
+		return nil, rpc.ConvertError(err, "failed to convert events", codes.Internal)
+	}
```

Comment:
you shouldn't need to convert the events since you're not returning these.
```suggestion
```

technically, the `ExecuteCallbacksTransactions` method requires CCF encoded events. so you could add an explicit check that actualEventEncodingVersion is CCF, but ENs have been returning CCF encoded events for a couple years now and the decoding will fail with an error anyway.